sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -3820,7 +3856,7 @@ public void testSelectDistinctWithStrlenFilter() throws Exception
                         .setGranularity(Granularities.ALL)
                         .setVirtualColumns(
                             expressionVirtualColumn("v0", "strlen(\"dim1\")", ValueType.LONG),
-                            expressionVirtualColumn("v1", "CAST(strlen(\"dim1\"), 'STRING')", ValueType.STRING)
+                            expressionVirtualColumn("v1", "CAST(CAST(strlen(\"dim1\"), 'STRING'), 'LONG')", ValueType.LONG)
```

Comment:
Hmm, this seems strange and sort of lame in terms of the produced expression, but i guess more correct in terms of the types involved for the SQL?

Reply:
Something should be collapsing these casts. Could you add a comment to the test that we're aware the behavior here is weird? Maybe a future developer will see that and it will inspire them.